VizNetworkApp/Requests/ApiNetworkRequest.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ApiNetworkRequest.swift
+//  NetworkApp
+//
+//  Created by Ran Helfer on 27/07/2021.
+//
+
+import Foundation
+
+class ApiNetworkRequest<APIResource: ApiResource> : NetworkRequest {
+    
+    typealias ModelType = APIResource.ModelType
+    
+    let failedToRetrieveUrlFromApiResource = "failedToRetrieveUrlFromApiResource"
+    var apiResource: APIResource
+    private let transport: NetworkTransport
```

Comment:
please change to weak var to make it harder to retain cycle.
if you need to perform checks before calling the transport, and maybe even error because it might not be there, it's better than taking a chance with retain cycles.
have an error like "transport missing on load" to the request if the transport weak ref is gone before you try loading.

Reply:
I think we should have weak var, or unowned, only for the case where we are using .shared. 

Since for other use cases, it makes sense to have a strong reference to the injected transport. Such as tests or other usages. 

What do you think ?

Reply:
So basically in order to solve this issue we can do something like this:

```
    static private let sharedInstance = NetworkTransporter()
    unowned static private(set) var shared = sharedInstance

```

at `NetworkTransporter` - please let me know what you think